agent/consul/state/connect_ca_test.go

```diff
@@ -79,11 +77,11 @@ func TestStore_CAConfigCAS(t *testing.T) {
 	}
 
 	// Do another CAS, this time with the correct index
-	ok, err = s.CACheckAndSetConfig(2, 1, &structs.CAConfiguration{
+	err = s.CACheckAndSetConfig(2, 1, &structs.CAConfiguration{
 		Provider: "static",
 	})
-	if !ok || err != nil {
-		t.Fatalf("expected (true, nil), got: (%v, %#v)", ok, err)
+	if err != nil {
+		t.Fatalf("expected (true, nil), got: (%#v)", err)
```

Comment:
Perhaps replace this if with `require.NoError(t, err)` instead? I was going to point out the message is a little wrong now as we don't expect a bool, so we should at least change that but I'd probably just move it to `require` at this point. Up to you though it might be odd to have a mix of assertion styles in this test 🤷 .

Reply:
yes a `require.NoError(t, err)` is definitely nicer and we are converting as much as possible to use testify require. So, I will convert the whole test.